tests/integrations/arq/test_arq.py

```diff
@@ -105,36 +127,69 @@ async def division(_, a, b=0):
 
     division.__qualname__ = division.__name__
 
-    pool, worker = init_arq([division])
+    cron_func = async_partial(division, a=1, b=int(not job_fails))
+    cron_func.__qualname__ = division.__name__
+
+    cron_job = cron(cron_func, minute=0, run_at_startup=True)
+
+    pool, worker = init_arq(functions_=[division], cron_jobs_=[cron_job])
 
     events = capture_events()
 
     job = await pool.enqueue_job("division", 1, b=int(not job_fails))
     await worker.run_job(job.job_id, timestamp_ms())
 
-    if job_fails:
-        error_event = events.pop(0)
-        assert error_event["exception"]["values"][0]["type"] == "ZeroDivisionError"
-        assert error_event["exception"]["values"][0]["mechanism"]["type"] == "arq"
+    loop = asyncio.get_event_loop()
+    task = loop.create_task(worker.async_run())
+    await asyncio.sleep(1)
```

Comment:
This is needed because the cron job automatically execute after the `worker` starts.

There's no way to actually wait for a cron job to finish other than adding some delay before asserting results.